lib/vmdb/console_methods.rb

```diff
@@ -43,7 +43,10 @@ def simulate_queue_worker(break_on_complete: false, quiet_polling: true)
       raise NotImplementedError, "not implemented in production mode" if Rails.env.production?
       loop do
         q = with_console_sql_logging_level(quiet_polling ? 1 : ActiveRecord::Base.logger.level) do
-          MiqQueue.where(MiqQueue.arel_table[:queue_name].not_eq("miq_server")).order(:id).first
+          MiqQueue.where.not(:queue_name => "miq_server")
+                  .where.not(:deliver_on => nil)
```

Comment:
seems like this change does not account for MiqQueue messages where `deliver_on` is `nil`

Reply:
Yeah, I updated this locally to this:

```diff
diff --git a/lib/vmdb/console_methods.rb b/lib/vmdb/console_methods.rb
index 3ddc2143f7..61f6050f30 100644
--- a/lib/vmdb/console_methods.rb
+++ b/lib/vmdb/console_methods.rb
@@ -41,9 +41,11 @@ module Vmdb
     # Development helper method for Rails console for simulating queue workers.
     def simulate_queue_worker(break_on_complete: false, quiet_polling: true)
       raise NotImplementedError, "not implemented in production mode" if Rails.env.production?
+      deliver_on = MiqQueue.arel_table[:deliver_on]
       loop do
         q = with_console_sql_logging_level(quiet_polling ? 1 : ActiveRecord::Base.logger.level) do
-          MiqQueue.where(MiqQueue.arel_table[:queue_name].not_eq("miq_server")).order(:id).first
+          MiqQueue.where(deliver_on.not_eq(nil).or(deliver_on.lteq(Time.now.utc)))
+                  .where.not(:queue_name => "miq_server").order(:id).first
         end
         if q
           puts "\e[33;1m\n** Delivering #{MiqQueue.format_full_log_msg(q)}\n\e[0;m"
```

Which prevents items with `deliver_on` in the future from being executed, but also prevents any other job from being run as well.  Still need to figure out the best option here, but I can't currently reuse any of the `MIQ_QUEUE_PEEK` or `MIQ_QUEUE_GET` code:

https://github.com/ManageIQ/manageiq/blob/32c744d1/app/models/miq_queue.rb#L313-L332
https://github.com/ManageIQ/manageiq/blob/32c744d1/app/models/miq_queue.rb#L246-L260

As neither will allow for the `where.not(:queue_name => "miq_server")` logic to work.

Though... maybe I could, but just fetching all of the different `queue_name` types, and just rejecting `miq_server` from that list.  I will give that a shot at some point today.

Reply:
I am an idiot... it should be `IS NULL`, not `IS NOT NULL`... probably what Oleg was telling me before.

Anyway, will push my modified change up shortly.